### DISCUSSION TITLE
How to disable url length validation

### DISCUSSION THREAD

**Discussion by cantaylancapraz** [linkerd/linkerd2#6116] [General]
Hi,

I send a request via linkerd and url length might be longer than 500.000 character. 

In this case, Linkerd don't send request to server and throws 400. When I check the logs, I see following logs.
`INFO ThreadId(01) outbound:accept{peer.addr=192.168.61.189:35418 target.addr=132.145.37.221:8002}: linkerd2_app_core::serve: Connection closed error=invalid URI 
`
I want to disable uri length validation. How can I do this?

**Comment by olix0r** [comment]
@cantaylancapraz 

The length limit is hardcoded in the HTTP library at [64KB](https://github.com/hyperium/http/blob/73c723edb4d16c92aef2a682cf43b00ce988a40a/src/uri/mod.rs#L144), which, it turns out, is still quite a bit larger than [most HTTP implementations support](https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers).

Generally speaking, it's not sustainable for HTTP servers to support URIs of extreme lengths like 500KB, as this would allow malicious clients to consume exorbitant resources, eventually forcing the server to run out of memory and be killed.

Have you considered using POST requests or another method for encoding data in HTTP payloads rather than in the URI?
